### ISSUE TITLE
The provided CROSSTOOL does not include proper support for dSYM generation for apple_binary

### ISSUE THREAD

**Issue by sergiocampama** [bazelbuild/bazel#4312]
Please provide the following information. The more we know about your system and use case, the more easily and likely we can help.

### Description of the problem / feature request / question:

When invoking `bazel build examples/ios/HelloWorldSwift  --apple_generate_dsym` on the rules_apple repository, the Linking action fails with the following message:

```
ERROR: /Users/kaipi/Development/Github/rules_apple/examples/ios/HelloWorldSwift/BUILD:18:1: output 'examples/ios/HelloWorldSwift/HelloWorldSwift.apple_binary.app.dSYM.temp.zip' was not created
ERROR: /Users/kaipi/Development/Github/rules_apple/examples/ios/HelloWorldSwift/BUILD:18:1: not all outputs were created or valid
```

I traced this to the provided crosstool missing the proper `DSYM_HINT_DSYM_PATH`, `DSYM_HINT_DSYM_BUNDLE_ZIP` and `DSYM_HINT_LINKED_BINARY` configurations for the `generate_dsym_file` feature.

Original issue: https://github.com/bazelbuild/tulsi/issues/38

### If possible, provide a minimal example to reproduce the problem:

Check out the https://github.com/bazelbuild/rules_apple and run the command above.

### Environment info
* Operating System:

macOS 10.13

* Bazel version (output of `bazel info release`):

release 0.8.1-homebrew

* If `bazel info release` returns "development version" or "(@non-git)", please tell us what source tree you compiled Bazel from; git commit hash is appreciated (`git rev-parse HEAD`):

###  Have you found anything relevant by searching the web?
(e.g. [StackOverflow answers](http://stackoverflow.com/questions/tagged/bazel),
[GitHub issues](https://github.com/bazelbuild/bazel/issues),
email threads on the [`bazel-discuss`](https://groups.google.com/forum/#!forum/bazel-discuss) Google group)

N/A

### Anything else, information or logs or outputs that would be helpful?
(If they are large, please upload as attachment or provide link).

N/A

**Comment by KieranLafferty** [comment]
Any known workaround for this at the moment?

**Comment by dolphineye** [comment]
We have the same issue in my organisation. We haven't found any workaround for now. The problem is not fixed with the latest 0.10.0-homebrew version. Any news? Thanks for your help :)

**Comment by sergiocampama** [comment]
This should be fixed at head, probably will take a while to get rolled out.
